[Bug] test_video_access error for python 3.7/3.8 starting 2nd July

# 🐛 Bug

CI `test_video_access` started to report errors since yesterday PM for python 3.7 and 3.8. It was OK when Bob ran it yesterday morning and for other recent tests. Not sure what has been changed in between on the system or elsewhere. The tests can pass on python 3.6. The error is `TypeError` and seems to be associated with `pickle.load` below

```
tests/loaddata/test_video_access.py:101: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
kale/loaddata/video_access.py:254: in get_train
    n_classes=self._n_classes,
kale/loaddata/video_datasets.py:120: in __init__
    test_mode,
kale/loaddata/videos.py:149: in __init__
    self._parse_list()
kale/loaddata/video_datasets.py:124: in _parse_list
    self.video_list = [VideoRecord(x, self.img_path) for x in list(self.make_dataset())]
kale/loaddata/video_datasets.py:146: in make_dataset
    input_file = pickle.load(input_file)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   TypeError: __cinit__() takes at least 2 positional arguments (0 given)

pandas/_libs/internals.pyx:572: TypeError
```

Some related searches
https://github.com/h5py/h5py/issues/1092
https://stackoverflow.com/questions/34569664/typeerror-cinit-takes-exactly-8-positional-arguments-3-given-when-loa
https://bugs.python.org/issue32696

## To reproduce

See https://github.com/pykale/pykale/runs/2978244183 or https://github.com/pykale/pykale/runs/2975724777 or rerun past successful tests

** Stack trace/error message **
```
=================================== FAILURES ===================================
_ test_get_source_target[class_subset0-max-natural-0.1-EPIC;8;epic_D1_train.pkl;epic_D1_test.pkl-EPIC;8;epic_D1_train.pkl;epic_D1_test.pkl] _

source_cfg = 'EPIC;8;epic_D1_train.pkl;epic_D1_test.pkl'
target_cfg = 'EPIC;8;epic_D1_train.pkl;epic_D1_test.pkl', val_ratio = 0.1
weight_type = 'natural', datasize_type = 'max'
testing_cfg = CfgNode({'DATASET': CfgNode({'ROOT': '/home/runner/work/tests/test_data/video_test_data/', 'IMAGE_MODALITY': 'joint', ...R_TRAINLIST': 'epic_D1_train.pkl', 'TAR_TESTLIST': 'epic_D1_test.pkl', 'WEIGHT_TYPE': 'natural', 'SIZE_TYPE': 'max'})})
class_subset = [1, 3, 8]

    @pytest.mark.parametrize("source_cfg", SOURCES)
    @pytest.mark.parametrize("target_cfg", TARGETS)
    @pytest.mark.parametrize("val_ratio", VAL_RATIO)
    @pytest.mark.parametrize("weight_type", WEIGHT_TYPE)
    @pytest.mark.parametrize("datasize_type", DATASIZE_TYPE)
    @pytest.mark.parametrize("class_subset", CLASS_SUBSETS)
    def test_get_source_target(source_cfg, target_cfg, val_ratio, weight_type, datasize_type, testing_cfg, class_subset):
        source_name, source_n_class, source_trainlist, source_testlist = source_cfg.split(";")
        target_name, target_n_class, target_trainlist, target_testlist = target_cfg.split(";")
        n_class = eval(min(source_n_class, target_n_class))
    
        # get cfg parameters
        cfg = testing_cfg
        cfg.DATASET.SOURCE = source_name
        cfg.DATASET.SRC_TRAINLIST = source_trainlist
        cfg.DATASET.SRC_TESTLIST = source_testlist
        cfg.DATASET.TARGET = target_name
        cfg.DATASET.TAR_TRAINLIST = target_trainlist
        cfg.DATASET.TAR_TESTLIST = target_testlist
        cfg.DATASET.WEIGHT_TYPE = weight_type
        cfg.DATASET.SIZE_TYPE = datasize_type
    
        download_file_by_url(
            url=url,
            output_directory=str(Path(cfg.DATASET.ROOT).parent.absolute()),
            output_file_name="video_test_data.zip",
            file_format="zip",
        )
    
        # test get_source_target
        source, target, num_classes = VideoDataset.get_source_target(
            VideoDataset(source_name), VideoDataset(target_name), seed, cfg
        )
    
        assert num_classes == n_class
        assert isinstance(source, dict)
        assert isinstance(target, dict)
        assert isinstance(source["rgb"], VideoDatasetAccess)
        assert isinstance(target["rgb"], VideoDatasetAccess)
        assert isinstance(source["flow"], VideoDatasetAccess)
        assert isinstance(target["flow"], VideoDatasetAccess)
    
        # test get_train & get_test
>       assert isinstance(source["rgb"].get_train(), torch.utils.data.Dataset)

tests/loaddata/test_video_access.py:101: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
kale/loaddata/video_access.py:254: in get_train
    n_classes=self._n_classes,
kale/loaddata/video_datasets.py:120: in __init__
    test_mode,
kale/loaddata/videos.py:149: in __init__
    self._parse_list()
kale/loaddata/video_datasets.py:124: in _parse_list
    self.video_list = [VideoRecord(x, self.img_path) for x in list(self.make_dataset())]
kale/loaddata/video_datasets.py:146: in make_dataset
    input_file = pickle.load(input_file)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   TypeError: __cinit__() takes at least 2 positional arguments (0 given)

pandas/_libs/internals.pyx:572: TypeError
----------------------------- Captured stdout call -----------------------------
Downloading https://github.com/pykale/data/raw/main/video_data/video_test_data.zip to /home/runner/work/tests/test_data/video_test_data.zip
Extracting /home/runner/work/tests/test_data/video_test_data.zip to /home/runner/work/tests/test_data
----------------------------- Captured stderr call -----------------------------
```

## Expected Behaviour

I've rerun 10+ times in the past 18 hours but all failed

## Environment

See GitHub actions

_It was successful in PR #174 before merging (otherwise, merging won't be possible) so the problem may be system or environment dependent._

My suggestion is to consider whether the Type Error is genuine:
```
    input_file = pickle.load(input_file)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   TypeError: __cinit__() takes at least 2 positional arguments (0 given)

pandas/_libs/internals.pyx:572: TypeError
```

Also, do we really need **_150_** tests for video_access? I feel that there are many unnecessary tests here.

--- a/setup.py
+++ b/setup.py
@@ -12,6 +12,7 @@ from setuptools import find_packages, setup
 # python -m pip install torch-cluster torch-scatter torch-sparse torch-spline
 requirements = [
     "numpy>=1.18.0",
+    "pandas<=1.2.5",
     "pytorch-lightning",
     "scikit-image>=0.16.2",
     "scikit-learn>=0.23.2",

